Doc tests (#743)
STORM-IRIT/Radium-Engine: doc/developer/develWithRadium.md

As a Radium contributor, and in most cases, you will have to work only with CTest, and let the CI system run your tests automatically.
In this section we describe how to write a new test, how to insert it in the testing suite, and how to validate your results on your computer and the CI systems.

## How to write tests

There is two kind of tests, integration test and unit test. Tests are run with the `check` target and it's variant (`check_verbose`, `test`, `run_unittests`, `run_unittests_verbose`)

> * integration test are based on program return values, using the ctest suite. The tests are set in `tests/integration/CMakeLists.txt`, see `add_test` [documenation](https://cmake.org/cmake/help/latest/command/add_test.html).

Reviewer: ```suggestion
* integration test are based on program return values, using the ctest suite. The tests are set in `tests/integration/CMakeLists.txt`, see `add_test` [documentation](https://cmake.org/cmake/help/latest/command/add_test.html).
```

>
> Source: [https://www.toptal.com/qa/how-to-write-testable-code-and-why-it-matters](https://www.toptal.com/qa/how-to-write-testable-code-and-why-it-matters)

To add a unit test, 
create the test file in tests/unittest/LIB/mytestfile.cpp where LIB correspond to the library you are testing (Core, Engine, Gui, IO).

> Add this file to the `tests/unittest/CMakeLists.txt` to the list of sources of `add_executable(unittests ...`.

Reviewer: ```suggestion
Edit `tests/unittest/CMakeLists.txt` and add your test to the list of sources of `add_executable(unittests ...`.
```
